src/ray/util/process.cc

```diff
@@ -76,40 +95,25 @@ class ProcessFD {
 
   // Fork + exec combo. Returns -1 for the PID on failure.
   static ProcessFD spawnvpe(const char *argv[], std::error_code &ec, bool decouple,
-                            const std::map<std::string, std::string> &env) {
+                            const ProcessEnvironment &env) {
     ec = std::error_code();
     intptr_t fd;
     pid_t pid;
-#ifdef _WIN32
-    LPTCH env_strings = GetEnvironmentStrings();
```

Comment:
@mehrdadn Can you explain why we don't use `GetEnvironmentStrings`? According to https://stackoverflow.com/questions/54370749/is-environ-available-on-windows, `environ` is deprecated on Windows.

Reply:
It's more consistent, and it factors out the logic and shares code with POSIX. Also, if you had stuck with `GetEnvironmentStrings`, you would've needed to use `LPCH` and `GetEnvironmentStringsA` instead of `GetEnvironmentStrings` because the latter can use `WCHAR` instead of `CHAR`.

The deprecation is irrelevant here, at least as far as Windows goes. The problem is the same as on UNIX systems: `environ` is not thread-safe, so if you modify environment variables in another thread, you can't read it here without locking. Microsoft's C runtime provides methods to lock the environment table and copy the data atomically, so they encourage you to use that instead. But it's only an issue if you're changing environment variables in another thread. And if you are (which I don't believe we are), it'd be just as buggy on POSIX systems. But given we're not, there's no reason to avoid it in Windows.